Update to 1.13

piratjsk opened the pull request:
- updated spigot api version to 1.13
- code adjusted to api changes
- mcmmo added as maven dependency using jitpack.io
- reflections in IdentityUtils replaced with calls to bukkit api
- fixed NullPointerException when using `/mf shop <player>` command while shop is disabled

After this changes plugin won't be working with older (below 1.13) spigot versions.

While it seems to work without any major bugs on 1.13, there still need to be some more testing done before releasing, so if someone would like to help and test this build, here is download link: https://www.dropbox.com/s/4dcy8qw23wjz79b/morefish-2.2-SNAPSHOT.jar?dl=0.

piratjsk committed:
update to 1.13

piratjsk committed:
add mcmmo as maven dependency using jitpack.io repo

piratjsk committed:
use bukkit api instead of reflections

piratjsk committed:
update item names to 1.13

piratjsk committed:
fix nullpointerexception when using /mf shop player and shop is disabled

kaimokene commented:
Hello. The auto fishing contest doesnt seem to work, also the fish shop says in config that its enabled while ingame it says its disabled. also the fish potion effects wont work since the syntax has changed for the command. I know the potion effect thing can be changed manually but id rather have it auto generate with right syntax. Currently thats all bugs that i have encountered there is a error that literally says only null in console when starting server but thats about it.

piratjsk commented:
> [...] Fish shop says in config that its enabled while ingame it says its disabled. [...]

Fish shop requires Vault and an economy plugin to work. If one of them isn't installed then enabling shop in config will take no effect. I've added some warnings about that on plugin startup in [the latest version](https://github.com/piratjsk/Rby/releases/tag/mf-v2.2.1) of my fork.  
If you have installed Vault and an economy plugin and this problem is still present please provide name and version of economy plugin you are using.
> [...] also the fish potion effects wont work since the syntax has changed [...]

I already fixed this (and few other things) in my fork, but didn't get to updating this pull request.  
It also seems that original author is not interested in this plugin at this moment so I'll be just maintaining my fork.

> [...] there is a error that literally says only null in console when starting server [...]

This also should be fixed in [the latest version](https://github.com/piratjsk/Rby/releases/latest) of my fork.


So, to sum up:

- More Fish
  - latest download of my updated to 1.13 fork can be found at: https://github.com/piratjsk/Rby/releases/latest
  - it also includes some bugfixes
  - I plan to maintain it (keep compatible with latest spigot versions, fix bugs) until release of Rby
  - I will not be adding new features to this version
  - code of this version can be found in [mf-2.x branch](https://github.com/piratjsk/Rby/tree/mf-2.x) of [piratjsk/Rby](https://github.com/piratjsk/Rby)

- Rby
  - in meantime I will be developing new, heavily refactored version that will be released under new name Rby
  - this version will contain some new features and improvments to existing ones
  - my main focus is to make it more configurable and easily extensible by other plugins
  - code of this version can be found in [master branch](https://github.com/piratjsk/Rby/tree/master) of [piratjsk/Rby](https://github.com/piratjsk/Rby)
